sdcm/utils/replication_strategy_utils.py

```diff
@@ -1,45 +1,69 @@
 import ast
+import logging
 import re
 
 from contextlib import ContextDecorator
-from typing import Callable, Dict
+from typing import Callable, Dict, TYPE_CHECKING
 
-from sdcm.cluster import BaseNode
 from sdcm.utils.cql_utils import cql_quote_if_needed
+if TYPE_CHECKING:
+    from sdcm.cluster import BaseNode
+
+LOGGER = logging.getLogger(__name__)
 
 
 class ReplicationStrategy:  # pylint: disable=too-few-public-methods
 
     @classmethod
     def from_string(cls, replication_string):
-        replication_value = re.search(r".*replication[\s]*=[\s]*(\{.*\})", replication_string, flags=re.IGNORECASE)
+        # To solve the problem when another curly braces were added (tablets related).
+        # Example:
+        # CREATE KEYSPACE scylla_bench WITH replication = {'class': 'org.apache.cassandra.locator.SimpleStrategy',
+        # 'replication_factor': '1'} AND durable_writes = true AND tablets = {'enabled': false};
+        LOGGER.debug("Analyze replication string '%s'", replication_string)
+        if not (replication_value := re.search(r".*replication[\s]=[\s](\{.*'\d+'\})", replication_string, flags=re.IGNORECASE)):
+            # For syntax: CREATE KEYSPACE scylla_bench WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}
+            replication_value = re.search(
+                r".*replication[\s]=[\s](\{.*:[\s]\d+\})", replication_string, flags=re.IGNORECASE)
+
         strategy_params = ast.literal_eval(replication_value[1])
         strategy_class = strategy_params.pop("class")
         for class_ in replication_strategies:
-            if strategy_class == class_.class_:
+            # To cover short and long class name, like:
+            #   - SimpleStrategy
+            #   - org.apache.cassandra.locator.SimpleStrategy
+            if strategy_class.endswith(class_.class_):
                 return class_(**strategy_params)
         raise ValueError(f"Couldn't find such replication strategy: {replication_value}")
 
     @classmethod
-    def get(cls, node: BaseNode, keyspace: str):
+    def get(cls, node: 'BaseNode', keyspace: str):
         create_ks_statement = node.run_cqlsh(f"describe {keyspace}").stdout.splitlines()[1]
         return ReplicationStrategy.from_string(create_ks_statement)
 
-    def apply(self, node: BaseNode, keyspace: str):
+    def apply(self, node: 'BaseNode', keyspace: str):
```

Comment:
can this change be removed if you used `if TYPE_CHECKING:`?

Reply:
no. Pylint fails with:
```
Using variable 'BaseNode' before assignment (used-before-assignment)
```